components/Templates/includes/auto-template/Pods_Templates_Auto_Template_Front_End.php

```diff
@@ -19,100 +28,117 @@ public function __construct() {
 			add_action( 'wp', array( $this, 'set_frontier_style_script' ) );
 		}
 
+		// Setup initial hooks.
 		add_action( 'template_redirect', array( $this, 'hook_content' ) );
+		// Setup hooks after each new post in the loop.
+		add_action( 'the_post', array( $this, 'hook_content' ) );
 
+		// Setup comment hooks.
+		add_action( 'template_redirect', array( $this, 'maybe_hook_content' ) );
 	}
 
 	/**
-	 * Get the filter used for a Pod.
-	 * @since  1.7.2
-	 * @param  string  $current_post_type
-	 * @param  array   $possible_pods
-	 * @return string
+	 * Add hooks for output.
+	 *
+	 * @since 2.6.6
 	 */
-	public function get_pod_filter( $current_post_type = '', $possible_pods = array() ) {
-		$filter = 'the_content';
+	public function hook_content(){
+		$filter = $this->get_pod_filter();
 
-		if ( ! $current_post_type ) {
-			// get the current post type
-			$current_post_type = $this->current_post_type();
+		/**
+		 * Allows plugin to append/replace the_excerpt.
+		 *
+		 * Default is false, set to true to enable.
+		 */
+		if ( ! defined( 'PFAT_USE_ON_EXCERPT' ) ) {
+			define( 'PFAT_USE_ON_EXCERPT', false );
 		}
 
-		if ( ! $possible_pods ) {
-			//now use other methods in class to build array to search in/ use
-			$possible_pods = $this->auto_pods();
+		$this->filtered_content[ $filter ] = 10.5;
+
+		if ( PFAT_USE_ON_EXCERPT ) {
+			$this->filtered_content['the_excerpt'] = 10;
 		}
 
-		//check if $current_post_type is the key of the array of possible pods
-		if ( isset( $possible_pods[ $current_post_type ] ) ) {
-			$this_pod = $possible_pods[ $current_post_type ];
+		$this->install_hooks();
+	}
 
-			if ( is_singular( $current_post_type ) ) {
-				$filter = pods_v( 'single_filter', $this_pod, $filter, true );
-			} elseif ( is_post_type_archive( $current_post_type ) ) {
-				$filter = pods_v( 'archive_filter', $this_pod, $filter, true );
-			} elseif ( is_home() && $current_post_type === 'post'  ) {
-				$filter = pods_v( 'archive_filter', $this_pod, $filter, true );
-			} elseif ( is_tax( $current_post_type )  ) {
-				$filter = pods_v( 'archive_filter', $this_pod, $filter, true );
-			}
+	/**
+	 * Check whether to add the hook for comment templates.
+	 *
+	 * @since 2.7.23
+	 */
+	public function maybe_hook_content() {
+		$possible_pods = $this->auto_pods();
+
+		if ( isset( $possible_pods['comment'] ) ) {
+
+			// Comment auto templates.
+			add_filter( 'comment_text', array( $this, 'front' ), 99, 2 );
```

Comment:
:no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 3ac3e2a54f5ebb9f54534987972062e8</sub>